packages/viz/src/lib/style/helpers/utils.ts

```diff
@@ -0,0 +1,84 @@
+import { CartoStylingError, stylingErrorTypes } from '../../errors/styling-error';
+import { getColorPalette } from '../palettes';
+
+export function validateParameters(
+  featureProperty: string,
+  colors: string[] | string,
+  lengthComparisonFn: () => {}
+) {
+  if (!featureProperty) {
+    throw new CartoStylingError('Feature property is missing', stylingErrorTypes.PROPERTY_MISSING);
+  }
+
+  const lengthMismatch = lengthComparisonFn();
+  const colorsIsNotString = typeof colors !== 'string';
+
+  if (colorsIsNotString && lengthMismatch) {
+    throw new CartoStylingError(
+      'Numeric values for ranges length and color length do not match',
+      stylingErrorTypes.PROPERTY_MISMATCH
+    );
+  }
```

Comment:
Perhaps we can add another validation here, if colors === 'string' (palette name), that the palette exists in cartocolors., reusing code from 'getColors' method

Reply:
Honestly, I prefer to keep them separated, because here we are like validating the type/shape, rather than if they're valid or not.

If we were to add that kind of validation, we'd need to add some other ones like checking if the HEX string is valid, and I prefer not to put them here.

Reply:
fair point